force-app/main/default/lwc/generateRecordInputForCreate/generateRecordInputForCreate.js

```diff
@@ -0,0 +1,76 @@
+import { LightningElement, wire } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import {
+    createRecord,
+    getRecordCreateDefaults,
+    generateRecordInputForCreate
+} from 'lightning/uiRecordApi';
+import { reduceErrors } from 'c/ldsUtils';
+import ACCOUNT_OBJECT from '@salesforce/schema/Account';
+import NAME_FIELD from '@salesforce/schema/Account.Name';
+import AREANUMBER_FIELD from '@salesforce/schema/Account.AreaNumber__c';
+
+export default class GenerateRecordInputForCreate extends LightningElement {
+    areaNumber;
+    areaNumberField = AREANUMBER_FIELD;
+    areaNumberCreateable;
+    error;
+    name;
+    nameField = NAME_FIELD;
+    recordInput;
+
+    @wire(getRecordCreateDefaults, { objectApiName: ACCOUNT_OBJECT })
+    loadAccountCreateDefaults({ data, error }) {
+        if (data) {
+            const accountObjectInfo =
+                data.objectInfos[ACCOUNT_OBJECT.objectApiName];
+            const recordDefaults = data.record;
+            // Creates an account record input with default field values
+            this.recordInput = generateRecordInputForCreate(
```

Comment:
I'm wondering if this even works as expected. Even when removing the `edit` permissions on the SysAdmin profile _and_ in the permissionset I get `Area_Number__c` back from `generateRecordInputForCreate`.

Reply:
This is because of the sys admin profile, that has the "Edit Read Only Fields" permission activated by default. Try out removing read permissions for the field. Removing them in the permission set should be enough as you shouldn't have permissions in the profile unless you've explicitly changed it.